File: Backend/tweetApp/src/main/java/com/tweetApp/tweetApp/service/UserService.java
package com.tweetApp.tweetApp.service;

import java.util.List;

import com.tweetApp.tweetApp.DTO.RegisterDTO;
import com.tweetApp.tweetApp.DTO.ResetPasswordDTO;
import com.tweetApp.tweetApp.model.Register;

public interface UserService {

	RegisterDTO newRegistration(RegisterDTO registerdto);
	RegisterDTO loginUser(RegisterDTO registerdto);
	boolean resetPassword(String userName,ResetPasswordDTO resetPasswordDTO);
	List<Register> getAllUsers();
}

File: Frontend/frontend/src/app/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../environments/environment';
import { Register } from './login/register';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  // baseUrl = `http://localhost:8090/api/v1.0/tweets`
  constructor(private httpClient:HttpClient) { }

  loginUser(user:Register){
   return this.httpClient.post(environment.baseUrl + '/login', user);
  }

  getAllUsers(){
    return this.httpClient.get(environment.baseUrl  + '/users/all');
  }

  resetPassword(resetForm:any){
    let body = {
   userName : resetForm.value.userName,
   newPassword : resetForm.value.password
    }
    console.log(body)
    return this.httpClient.put(environment.baseUrl+ '/'+resetForm.value.userName+'/forgot',body);
  }

  registerUser(signupForm){
    let body={
      firstName:signupForm.value.firstName,
      lastName:signupForm.value.lastName,
      email:signupForm.value.email,
      userName:signupForm.value.userName,
      password:signupForm.value.password,
      contactNum:signupForm.value.contactNum
    }
    return this.httpClient.post(environment.baseUrl + "/register",body);
  }
}
